### PR TITLE
Add token top transactions metric

### PR DESCRIPTION
**Pull request by spiderjako**

### CODE REVIEW

File: src/functions/blockchainFunctions.js [hunk 1]
```diff
@@ -146,3 +146,19 @@ function exchangeFundsFlow_ (slug, from, to) {
...
+function tokenTopTransactions_ (slug, from, to, limit) {
+  var results = getApiClient_().fetchTokenTopTransactions(slug, from, to, limit)
+  assertHasData_(results)
+  var headers = ['Date', 'From', 'To', 'TRX Hash', 'TRX Value']
+
+  return [headers].concat(results.tokenTopTransactions.map(function (result) {
+    return [
+      formatDatetimeField_(result.datetime),
+      result.fromAddress,
+      result.toAddress,
```
Comment (IvanIvanoff) [inline src/functions/blockchainFunctions.js]: This should be separated into address and isExchange fields

File: doc/sheet_functions.md [hunk 2]
```diff
@@ -121,6 +121,9 @@ Currently only ETH is supported.</p>
 <dt><a href="#SAN_DAILY_CLOSING_PRICE">SAN_DAILY_CLOSING_PRICE(projectSlug, day)</a> ⇒ <code>number</code></dt>
 <dd><p>Returns the closing price for a given day.</p>
 </dd>
+<dt><a href="#SAN_TOKEN_TOP_TRANSACTIONS">SAN_TOKEN_TOP_TRANSACTIONS(projectSlug, from, to, limit)</a> ⇒ <code>Array</code></dt>
+<dd><p>Top transactions for the token of a given project.</p>
```
Comment (kalkov) [inline doc/sheet_functions.md]: "Top token transactions for a given slug" sounds better to me

File: test/functions/blockchainFunctionsTest.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+require('../helper.js')
+
+const slug = 'santiment'
+const from = new Date(2019, 0, 1, 0, 0, 0)
```
Comment (kalkov) [inline test/functions/blockchainFunctionsTest.js]: Same as previous PRs, dates doesn't correspond to response.

File: src/functions/blockchainFunctions.js [hunk 4]
```diff
@@ -146,3 +146,21 @@ function exchangeFundsFlow_ (slug, from, to) {
     ]
   }))
 }
+
+function tokenTopTransactions_ (slug, from, to, limit) {
+  var results = getApiClient_().fetchTokenTopTransactions(slug, from, to, limit)
+  assertHasData_(results)
+  var headers = ['Date', 'From Address', 'From Is Exchange', 'To Address', 'To Is Exchange', 'TRX Hash', 'TRX Value']
```
Comment (kalkov) [inline src/functions/blockchainFunctions.js]: TRX Hash -> Transaction Hash
TRX Value -> Transacrtion Value

File: test/functions/blockchainFunctionsTest.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+const to = new Date(2019, 0, 3, 0, 0, 0)
+const limit = 5
```
Comment (kalkov) [inline test/functions/blockchainFunctionsTest.js]: Also limit is 5, results - 2

File: test/integration/sanTokenTopTransactions.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  testHandlesNullData(
+    'fetchTokenTopTransactions',
+    san.SAN_TOKEN_TOP_TRANSACTIONS,
+    slug,
+    from,
+    to,
+    limit)
+
+  it('has proper headers', () => {
+    const expectedHeaders = ['Date', 'From Address', 'From Is Exchange', 'To Address', 'To Is Exchange', 'TRX Hash', 'TRX Value']
```
Comment (kalkov) [inline test/integration/sanTokenTopTransactions.js]: This is too long, need to scroll horizontally.

File: test/setup.js [hunk 7]
```diff
@@ -20,6 +20,8 @@ const days = eachDay(from, subDays(to, 1)) // last day should not be included (h
 const historicDataTo = subDays(endOfYesterday(), 200)
 const historicDataFrom = startOfDay(subDays(historicDataTo, 205))
 
+const limit = 5
```
Comment (kalkov) [inline test/setup.js]: You  can remove this limit, it is used in only one test.

File: doc/sheet_functions.md (hunk 2 above)
Comment (spiderjako) [reply doc/sheet_functions.md]: You're right, I corrected it now

File: src/functions/blockchainFunctions.js (hunk 4 above)
Comment (spiderjako) [reply src/functions/blockchainFunctions.js]: It should be okay now

File: test/functions/blockchainFunctionsTest.js (hunk 5 above)
Comment (spiderjako) [reply test/functions/blockchainFunctionsTest.js]: I've just removed the limit and replaced the limit in the test with 5

File: test/integration/sanTokenTopTransactions.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+  testFieldTypes(results, expected)
+  testHandlesNullData(
+    'fetchTokenTopTransactions',
+    san.SAN_TOKEN_TOP_TRANSACTIONS,
+    slug,
+    from,
+    to,
+    5)
+
```
Comment (kalkov) [inline test/integration/sanTokenTopTransactions.js]: Should we also have a test for `testHistoricDataIsForbidden`?

Comment (spiderjako) [reply test/integration/sanTokenTopTransactions.js]: No, I've tested the function in sanpy with dates from 2018 and it isn't a premium metric

Comment (kalkov) [reply test/integration/sanTokenTopTransactions.js]: we have decided that it's not needed

File: src/functions/blockchainFunctions.js [hunk 9]
```diff
@@ -160,3 +160,28 @@ function historicalBalance_ (slug, from, to, address) {
...
+  ]
+  return [headers].concat(results.tokenTopTransactions.map(function (result) {
+    return [
+      formatDatetimeField_(result.datetime),
+      result.fromAddress.address,
+      result.fromAddress.isExchange,
+      result.toAddress.address,
+      result.toAddress.isExchange,
+      result.trxHash,
+      result.trxValue
```
[the remark this answers is not in the document]
Comment (spiderjako) [reply src/functions/blockchainFunctions.js]: I think, that I've tested it out and there isnt much difference, but I will do it again, just to make sure

[the remark this answers is not in the document]
Comment (spiderjako) [reply src/functions/blockchainFunctions.js]: Yes, at least for trxValue there isn't any difference. Should I put it anyways?

[the remark this answers is not in the document]
Comment (kalkov) [reply src/functions/blockchainFunctions.js]: Yes, please

**Pull request merged**